src/player.cpp

```diff
@@ -512,17 +514,17 @@ void player::action_taken()
 void player::update_morale()
 {
     // Decay existing morale entries.
-    for( size_t i = 0; i < morale.size(); i++ ) {
+    for( size_t i = morale.size(); i --> 0; ) {
```

Comment:
Ahh, the `-->` operator, also known as "goes to". However, it does not exist. Please don't write it as if it were an actual single operator.

If you want to iterate over a container in reverse, use this idiom with iterators instead:
```C++
        for( auto iter = worn.rbegin(); iter != worn.rend(); ) {
            if( armor_absorb( elem, armor ) ) {
                iter = decltype(iter)( worn.erase( --iter.base() ) );
            } else {
                ++iter;
            }
        }
```
See it in action in `player::absorb_hit`, also has comments.

Or alternatively, which may be easiest to understand and read, use two loops. One for processing and one for erasing.
```C++
for( auto iter = rbegin(); iter != rend(); ++iter) {
    iter->proceed();
}
morale.erase( std::remove_if( morale.begin(), morale.end(), [](const morale_point &m) { return m.is_expired(); }, morale.end() );
```

Btw. do you even *need* to process the morale points in reverse? It's currently not done, and `morale_point::proceed` does not seem to use anything that may depend on previously proceeded morale points.